app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+class ExpensesController < ApplicationController
+  # respond_to :json
+
+  def index
+    @budget_id = params[:budget_plan_id]
+    unless @budget_id.blank?
+      begin
+        @plan = current_user.budget_plans.find(@budget_id)
+
+      rescue ActiveRecord::RecordNotFound => e
+        render(json: { error: 'Budget plan not found for this user.' }, status: :not_found)
+        return
+      end
+      if @plan
```

Comment:
```suggestion
```
this will always be true since you are `rescue'ing` and returning above

Reply:
sadly you are correct again lol